### ISSUE TITLE
For how long it would support bash on Windows?

### ISSUE THREAD

**Issue by kiwi0fruit** [conda-forge/git-feedstock#53]
git on win from anaconda doesn't ship bash but conda-forge git ships bash. And I guess this may one of the reasons why build number is 1000: when priority to conda-forge then git would be with bash.

One of my packages depends on bash on win. Is it reliable to depend on conda-forge git? or bash would be removed eventually from it?

**Comment by scopatz** [comment]
I don't think anyone is really looking to add bash on Windows right now

**Comment by kiwi0fruit** [comment]
But it's already there at the moment (git package on Windows contains bash)! So I'm curious if it would be removed or not. For example on Windows 7 conda build errored in cmd but worked fine in git bash (that os repackaged in conda forge package).

**Comment by scopatz** [comment]
Right, so it is a bit of a hack right at the moment. But if anyone wants to add windows support to the Bash feedstock, that would be great!

**Comment by kiwi0fruit** [comment]
it's not that easy if you want to compile from source and it's tricky if
you want to repackage git bash for windows. As the latter also contains git
and that would mess with git package.

so I guess the mentioned hack is the simplest solution. I was thinking
about how it could work other way and didn't come with a better solution.

But still it's not the best one as it may lead to the binaries
incompatibility.

>

**Comment by scopatz** [comment]
Right, no one said it was easy or fun :)

**Comment by msarahan** [comment]
Let me explain the discrepancy.  The package on defaults is a repackage of git4win's MinGit: https://github.com/git-for-windows/git/releases

The conda-forge package is the full git4win.  Defaults does not do the full git4win because it has subtle incompatibilities with msys2, which defaults also ships.  Defaults recommends using the "posix" metapackage, which brings in m2-bash and other things to give you bash.  It is a bit old at this point, but it works.  You can use the defaults posix package with the defaults git package.  You can't use the defaults posix package with the conda-forge git package.

**Comment by kiwi0fruit** [comment]
Wow. Thanks for clarifying this. By the way: was this incompatibilities
somehow put into dependencies or it's a metaknowledge that the package
developers should use?

**Comment by msarahan** [comment]
unfortunately, it's not baked into dep info in any way right now.  It should probably be some kind of mutex metapackage for things that depend on the two different core MSYS2 dll's (git4win is based on msys2, but not quite compatible).  It would probably be best to remove the bash from the conda-forge git package (sync the recipe with defaults), and then document how to install the "posix" package to get bash in the compatible way.

**Comment by kiwi0fruit** [comment]
can conda-forge packages depend on packages from msys2 channel?

**Comment by msarahan** [comment]
In theory, yes.  Should this git package depend on posix?  IMHO, no.  Usage preferences are not the same thing as package dependencies.  We can create metapackages that reflect user usage preference, but the git package itself should provide one thing: git.  Git4win set a bad precedent by bundling bash and all this stuff with it.  Yes, it's convenient for one thing, but what happens when it conflicts?

**Comment by kiwi0fruit** [comment]
Thanks! This actually answers my original question at the top of the issue.

**Comment by kiwi0fruit** [comment]
@msarahan By the way: what do you think about noach metapackages like this one: [posix-noarch](https://github.com/conda-forge/staged-recipes/pull/7649). It looks like a poor design but it would make CI so much faster!

**Comment by kiwi0fruit** [comment]
[git package redistrubutes some Microsoft copyrighted files + git post install script partially fails](https://github.com/conda-forge/git-feedstock/issues/56)

**Comment by kiwi0fruit** [comment]
@msarahan @scopatz please review [git-bash](https://github.com/conda-forge/staged-recipes/pull/7799) conda-forge package.
